nitransforms/resampling.py

```diff
@@ -233,6 +234,8 @@ def apply(
     if isinstance(spatialimage, (str, Path)):
         spatialimage = _nbload(str(spatialimage))
 
+    spatialimage = squeeze_image(spatialimage)
```

Comment:
Do we really want to squeeze? I think what we generally want is `new_shape = target_shape[:3] + source_shape[3:]`.

Reply:
Okay tests do think it's okay to squeeze: there are these potential cases:

| Input | Transform | Output | Example |
|:--:|:---:|:---:|:---:|
| 3D (including 4D singleton) | 3D | 3D | A T1w to MNI T1w transform |
| 4D | 4D | 4D (the number of timepoints in input and transform must match) | BOLD head-motion correction |
| 3D (including 4D singleton) | 4D | 4D (the 3D input resampled through each of the transformations) | A T1w to each of BOLD timepoints, including head-motion |
| 4D | 3D | 4D (the same transform applied to each timepoint) | Moving BOLD timepoints into T1w space, without considering head-motion |